module/VuFind/src/VuFind/View/Helper/Root/SchemaOrg.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * View helper for injecting schema.org metadata
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:developer_manual Wiki
+ */
+
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\HtmlAttributes;
+
+/**
+ * View helper for injecting schema.org metadata
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:developer_manual Wiki
+ */
+class SchemaOrg extends \Laminas\View\Helper\AbstractHelper
+{
+    /**
+     * Constructor
+     *
+     * @param HtmlAttributes $htmlAttributes HtmlAttributes view helper
+     * @param bool           $enabled        Is schema.org metadata enabled?
+     */
+    public function __construct(protected HtmlAttributes $htmlAttributes, protected bool $enabled = true)
+    {
+    }
+
+    /**
+     * Format schema.org attributes (if enabled).
+     *
+     * @param array $attributes HTML attributes (in key/value format)
+     *
+     * @return string
+     */
+    public function getAttributes(array $attributes): string
+    {
+        return $this->enabled ? ($this->htmlAttributes)($attributes) : '';
+    }
+
+    /**
+     * Get a schema.org tag (if enabled).
+     *
+     * @param string $tag        Tag name
+     * @param array  $attributes Tag attributes
+     *
+     * @return string
+     */
+    public function getTag(string $tag, array $attributes): string
```

Comment:
This is such a generally useful function, it's a shame to be stuck in this schema-specific class.  But can always be refactored later.  

Maybe worth noting in the comment that it generates specifically an opening (or void) tag and that any closing tag has to be added separately.

Reply:
We do have the [MakeTag view helper](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/View/Helper/Root/MakeTag.php) available for more general-purpose cases, though that covers slightly different scenarios than this logic.